message_definitions/v1.0/ardupilotmega.xml

```diff
@@ -1695,18 +1695,17 @@
       <field type="float" name="increment">OSD parameter increment.</field>
     </message>
     <message id="11037" name="OBSTACLE_DISTANCE_3D">
-      <wip/>
-      <!-- This message is work-in-progress and it can therefore change, and should NOT be used in stable production environments -->
```

Comment:
Kind of dubious removing the `wip` while also adding a field :-)

Fair illustration of where we need a separate message for describing the sensor vs supplying the data.

We really should have
```
<message id="11037" name="OBSTACLE_SENSOR_3D">
      <field type="uint32_t" name="time_boot_ms" units="ms">Timestamp (time since system boot).</field>
      <field type="uint8_t" name="sensor_id">Sensor id of the distance sensor.</field>
      <field type="uint8_t" name="sensor_type" enum="MAV_DISTANCE_SENSOR">Class id of the distance sensor type.</field>
      <field type="uint8_t" name="frame" enum="MAV_FRAME">Coordinate frame of reference.</field>
      <field type="uint8_t" name="orientation" enum="MAV_SENSOR_ORIENTATION">Direction the sensor faces. downward-facing: ROTATION_PITCH_270, upward-facing: ROTATION_PITCH_90, backward-facing: ROTATION_PITCH_180, forward-facing: ROTATION_NONE, left-facing: ROTATION_YAW_90, right-facing: ROTATION_YAW_270</field>
      <field type="float" name="min_distance" units="m">Minimum distance the sensor can measure.</field>
      <field type="float" name="max_distance" units="m">Maximum distance the sensor can measure.</field>
</message>
```

and then
```
     <message id="11038" name="OBSTACLE_DISTANCE_3D">
     <description>Obstacle located as a 3D vector.</description>
      <field type="uint32_t" name="time_boot_ms" units="ms">Timestamp (time since system boot).</field>
      <field type="uint8_t" name="sensor_id">Sensor id of the distance sensor.</field>
      <field type="uint16_t" name="obstacle_id" instance="true">Unique ID given to each obstacle so that its movement can be tracked. Use UINT16_MAX if object ID is unknown or cannot be determined.</field>
      <field type="float" name="x" units="m">X position of the obstacle.</field>
      <field type="float" name="y" units="m">Y position of the obstacle.</field>
      <field type="float" name="z" units="m">Z position of the obstacle.</field>
     </message>
```

We'd be sending 19 bytes of payload rather than 29.  Ship has probably sailed on this, however, as there's too much GCS support that would need to be fixed before we get Copter out the door.